EIPS/eip-5377.md

```diff
@@ -0,0 +1,117 @@
+---
+eip: 5377
+title: EIP-721 Consumable Extension - Errata
+description: Interface extension for EIP-721 consumer role
+author: Daniel Ivanov (@Daniel-K-Ivanov), George Spasov (@Perseverance), Pandapip1 (@Pandapip1)
+discussions-to: https://ethereum-magicians.org/t/EIP-4400-EIP721consumer-extension/7371
+status: Final
+type: Standards Track
+category: ERC
+created: 2021-10-30
+requires: 165, 721, 4400
+---
+
+## Abstract
+
+Identical to [EIP-4400](./eip-4400.md), but with a typo fixed.
+
+This specification defines standard functions outlining a `consumer` role for instance(s) of [EIP-721](./eip-721.md). An implementation allows reading the current `consumer` for a given NFT (`tokenId`) along with a standardized event for when an `consumer` has changed. The proposal depends on and extends the existing [EIP-721](./eip-721.md).
+
+## Motivation
+
+Many [EIP-721](./eip-721.md) contracts introduce their own custom role that grants permissions for utilising/consuming a given NFT instance. The need for that role stems from the fact that other than owning the NFT instance, there are other actions that can be performed on an NFT. For example, various metaverses use `operator` / `contributor` roles for Land (EIP-721), so that owners of the land can authorise other addresses to deploy scenes to them (f.e. commissioning a service company to develop a scene).
+
+It is common for NFTs to have utility other than ownership. That being said, it requires a separate standardized consumer role, allowing compatibility with user interfaces and contracts, managing those contracts.
+
+Having a `consumer` role will enable protocols to integrate and build on top of dApps that issue EIP-721 tokens. One example is the creation of generic/universal NFT renting marketplaces.
+
+Example of kinds of contracts and applications that can benefit from this standard are:
+- metaverses that have land and other types of digital assets in those metaverses (scene deployment on land, renting land / characters / clothes / passes to events etc.)
+- NFT-based yield-farming. Adopting the standard enables the "staker" (owner of the NFT) to have access to the utility benefits even after transferring his NFT to the staking contract
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Every contract compliant to the `EIP721Consumable` extension MUST implement the `IEIP721Consumable` interface. The **consumer extension** is OPTIONAL for EIP-721 contracts.
+
+```solidity
+/// @title EIP-721 Consumer Role extension
+///  Note: the EIP-165 identifier for this interface is `type(IEIP721Consumable).interfaceId`
```

Comment:
Generally I recommend including the interface id as an additional check for implementers.

Reply:
I don't like including it at all since it's explicitly stated in EIP-165, as well as it adds another thing that may become out of date and or might be wrong.